optuna/visualization.py

```diff
@@ -161,6 +158,113 @@ def _get_optimization_history_plot(study):
     return figure
 
 
+def plot_parallel_coordinate(study, params=[]):
+    # type: (Study, List[str]) -> None
+    """Plot the high-dimentional parameter relationships in a study.
+
+        Note that, If a parameter contains missing values, a trial with missing values is not
+        plotted.
+
+    Example:
+
+        The following code snippet shows how to plot the high-dimentional parameter relationships
+        inside Jupyter Notebook.
+
+        .. code::
+
+            import optuna
+
+            def objective(trial):
+                ...
+
+            study = optuna.create_study()
+            study.optimize(objective, n_trials=100)
+
+            optuna.visualization.plot_parallel_coordinate(study, params=['param_a', 'param_b'])
+
+    Args:
+        study:
+            A :class:`~optuna.study.Study` object whose trials are plotted for their objective
+            values.
+        params:
+            Parameter list to visualize. The default is all parameters.
+    """
+
+    _check_plotly_availability()
+    figure = _get_parallel_coordinate_plot(study, params)
+    figure.show()
+
+
+def _get_parallel_coordinate_plot(study, params=[]):
+    # type: (Study, List[str]) -> Figure
+
+    layout = go.Layout(
+        title='Parallel Coordinate Plot',
+    )
+
+    trials = [trial for trial in study.trials if trial.state == TrialState.COMPLETE]
+
+    if len(trials) == 0:
+        logger.warning('Your study does not have any completed trials.')
+        return go.Figure(data=[], layout=layout)
+
+    all_params = {p_name for t in trials for p_name in t.params.keys()}
+    if len(params) != 0:
+        for input_p_name in params:
+            if input_p_name not in all_params:
+                logger.warning('Parameter {} does not exist in your study.'.format(input_p_name))
+                return go.Figure(data=[], layout=layout)
+        all_params = set(params)
+    sorted_params = sorted(list(all_params))
+
+    dims = [{
+        'label': 'Objective Value',
+        'values': tuple([t.value for t in trials]),
+        'range': (min([t.value for t in trials]), max([t.value for t in trials]))
+    }]
+    for p_name in sorted_params:
+        values = []
+        for t in trials:
+            if p_name in t.params:
+                values.append(t.params[p_name])
+        is_categorical = False
+        try:
+            tuple(map(float, values))
+        except (TypeError, ValueError):
+            from collections import defaultdict
```

Comment:
Please move this `import` to the top of this file to follow Optuna's convention.